/*
 * @begin:license
 *
 * Copyright (c) 2015-2019, Benjamin Niemann <pink@odahoda.de>
 *
 * This program is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 2 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License along
 * with this program; if not, write to the Free Software Foundation, Inc.,
 * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
 *
 * @end:license
 */

syntax = "proto2";

package noisicaa.pb;

message StepSequencerStep {
  optional bool enabled = 1;
  optional float value = 2;
}

message StepSequencerChannel {
  enum Type {
    VALUE = 1;
    GATE = 2;
    TRIGGER = 3;
  }
  required Type type = 1;
  repeated uint64 steps = 2;
  optional float min_value = 3;
  optional float max_value = 4;
  optional bool log_scale = 5;
}

message StepSequencer {
  repeated uint64 channels = 1;
  optional bool time_synched = 2;
  optional uint32 num_steps = 3;
}
